### ISSUE TITLE
segmetation fault on an update 

### ISSUE THREAD

**Issue by jah-** [openstreetmap/Nominatim#1600]
Hi again, I ran the update process after update the osm2pgsql binary based on the comments in https://github.com/openstreetmap/Nominatim/issues/1598#issuecomment-564099725, the process was fine but after indexing the process crash by an segmentation fault of the binary exec.

`[1310013.141569] osm2pgsql[23280]: segfault at 48 ip 000055bdecb1aab0 sp 00007ffc810024e8 error 4 in osm2pgsql[55bdeca6c000+100000]
`

![Screenshot from 2019-12-10 18-34-32](https://user-images.githubusercontent.com/4587643/70571491-cdbf9180-1b7c-11ea-9e81-3a84a2a8f866.png)

Maybe I need to run again the update process or change the strategy, in that case I really appreciate any suggestion to fix this issue.

Thanks

**Comment by lonvia** [comment]
Please provide the exact osm2pgsql version you are using (sha1 of commit) and a complete backtrace.

**Comment by lonvia** [comment]
Never mind. I see the problem.

**Comment by jah-** [comment]
hi @lonvia 

The backtrace:

```
Reading symbols from osm2pgsql/osm2pgsql...done.
[New LWP 1945]
[New LWP 1953]
[New LWP 1955]
[New LWP 1946]
[New LWP 1954]
[New LWP 1957]
[New LWP 1956]
[New LWP 1952]
[New LWP 1959]
[New LWP 1958]
[New LWP 1950]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `./osm2pgsql/osm2pgsql -klas --number-processes 8 -C 2000 -O gazetteer -S /srv/n'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  std::vector<db_deleter_place_t::item_t, std::allocator<db_deleter_place_t::item_t> >::emplace_back<char&, long&> (this=0x40, 
    __args#0=__args#0@entry=@0x7fffeb831cef: 78 'N', __args#1=@0x7fffeb831cf0: 3634728992, __args#1@entry=@0x7fffeb831cf0: <optimized out>)
    at /usr/include/c++/7/bits/vector.tcc:98
98              if (this->_M_impl._M_finish != this->_M_impl._M_end_of_storage)
[Current thread is 1 (Thread 0x7f8428ec8780 (LWP 1945))]

```

**Comment by jah-** [comment]
hi, there's any suggestion to fix this problem? 
I can't see the issue, I change the freq to diff updates to:

 `@define('CONST_Replication_Url', 'https://planet.openstreetmap.org/replication/hour/');`

and ran the `./utils/update.php --init-updates` cmd again, and then the `./utils/update.php --import-osmosis-all` whit the same result:

```
nominatim@openstreetmap-testing-002:~/Nominatim-3.4.0/build$ ./utils/update.php --import-osmosis-all
Currently at sequence 62462 (28/10/2019 23:58:04) - 1 indexed
/srv/nominatim/.local/bin/pyosmium-get-changes --server https://planet.openstreetmap.org/replication/hour/ -o /srv/nominatim/Nominatim-3.4.0/build/osmosischange.osc -s 30 -I 62462
/srv/nominatim/Nominatim-3.4.0/build/osm2pgsql/osm2pgsql -klas --number-processes 1 -C 2000 -O gazetteer -S /srv/nominatim/Nominatim-3.4.0/settings/import-full.style -d nominatim -P 5432 --flat-nodes /srv/nominatim/Nominatim-3.4.0/data/planet.bin /srv/nominatim/Nominatim-3.4.0/build/osmosischange.osc
osm2pgsql version 1.2.0 (64 bit id space)

Allocating memory for dense node cache
Allocating dense node cache in one big chunk
Allocating memory for sparse node cache
Sharing dense sparse
Node-cache: cache=2000MB, maxblocks=32000*65536, allocation method=11
Mid: loading persistent node cache from /srv/nominatim/Nominatim-3.4.0/data/planet.bin
Mid: pgsql, cache=2000
Parsing gazetteer style file '/srv/nominatim/Nominatim-3.4.0/settings/import-full.style'.
Using projection SRS 4326 (Latlong)

Reading in file: /srv/nominatim/Nominatim-3.4.0/build/osmosischange.osc
Using XML parser.
Segmentation fault (core dumped)
Error executing osm2pgsql: 139
```

Thanks for your comments.

**Comment by lonvia** [comment]
Fixed in bc68ff1e43f2527039640b8642bbe533814a3561.

**Comment by jah-** [comment]
@lonvia thanks!
